parsing/ast_mapper.ml

```diff
@@ -419,27 +419,33 @@ module E = struct
         while_ ~loc ~attrs (sub.expr sub e1) (sub.expr sub e2)
     | Pexp_list_comprehension (e1, comp_types) -> 
       list_comprehension ~loc ~attrs (sub.expr sub e1)
-        (List.map (fun comp_types -> 
-          List.map (fun comp_type ->
-            match comp_type with
-            | From_to (p, e2, e3, dir) ->
-              From_to(sub.pat sub p, sub.expr sub e2,
-            sub.expr sub e3, dir)
-            | In (p, e2) -> 
-            In(sub.pat sub p, sub.expr sub e2)
-          )comp_types 
+        (List.map (fun {clauses=comp_types; guard}  -> 
+          let clauses =   
```

Comment:
Better to lift this out as a `map_comprehension` function. Keeping the cases in these functions, by lifting out the mapping of sub-components, makes them very regular and much easier to review for correctness.